### PR TITLE
Expose the new reporter options

### PR DESCRIPTION
**Pull request by kylekthompson**
This required a few bug fixes that I just lumped into this PR, but long story short:

- Uses the config file in this repo
- Fixes how the config file's reporters setting and cli flag reporters interact
- Exposes the github step summary and markdown summary reporters

### CODE REVIEW

**Comment by TAGraves** [comment]
It looks like the Captain Cloud link is broken in the output — is that an issue in Cloud, or here?

**Comment by kylekthompson** [comment]
That’s an issue with Captain Cloud. I’m planning on taking a closer look tomorrow!

**Comment by tonywok** [comment]
@kylekthompson https://github.com/rwx-research/cloud/pull/2133 should do it :)

File: .github/workflows/continuous_integration.yaml [hunk 1]
```diff
@@ -14,16 +14,12 @@ jobs:
             keep-derivations = true
             keep-outputs = true
       - run: CGO_ENABLED=0 LDFLAGS="-w -s" nix develop --command mage
-      - run: |
-          REPORT=true ./captain run \
-            --suite-id="captain-cli-ginkgo" \
-            --test-results=report.xml \
-            --fail-on-upload-error \
-            -- nix develop --command mage test
+      - run: ./captain run captain-cli-ginkgo
```
Comment (michaelglass) [inline .github/workflows/continuous_integration.yaml:17]: 😻

File: internal/reporting/.snapshots/Markdown Report produces a readable summary when cloud is enabled [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # `some-suite-id` Summary
 
-[🔗 View in Captain Cloud](https://example.com/deep_link/test_suite_summaries/some-suite-id/some%2Fbranch/abcdef113131)
+[🔗 View in Captain Cloud](https://example.com/deep_link/test_suite_summaries/some-suite-id/some/branch/abcdef113131)
```
Comment (michaelglass) [inline internal/reporting/.snapshots/Markdown Report produces a readable summary when cloud is enabled:3]: 👍

File: cmd/captain/run.go [hunk 3]
```diff
@@ -324,7 +324,7 @@ func bindRunCmdFlags(cfg Config, cliArgs CliArgs) Config {
 			suiteConfig.Output.Quiet = true
 		}
 
-		if cliArgs.reporters != nil {
+		if len(cliArgs.reporters) > 0 {
```
Comment (michaelglass) [inline cmd/captain/run.go:327]: I don't totally understand the implications here: is this fixing a bug? if so can you add a regression test

File: cmd/captain/run.go [hunk 4]
```diff
@@ -325,7 +325,7 @@ func bindRunCmdFlags(cfg Config, cliArgs CliArgs) Config {
 		}
 
 		if len(cliArgs.reporters) > 0 {
-			reporterConfig := make(map[string]string)
```
Comment (michaelglass) [inline cmd/captain/run.go:314]: this could also warrant a test? This kind of behavior can be a little finnicky

**Review by michaelglass — APPROVED** [review]

File: cmd/captain/run.go (hunk 3 above)
Comment (kylekthompson) [reply cmd/captain/run.go:327]: It's fixing a bug, but I can't add a regression test until we discuss https://rwxhq.slack.com/archives/C04J2EA6WCB/p1680816911816679 and https://rwxhq.slack.com/archives/C04J2EA6WCB/p1680817661697899 further.

I have a WIP PR here, but that even doesn't completely cover the problem because we don't have any tests covering the config file: https://github.com/rwx-research/captain/pull/17

TL;DR: reporters from the config file were never respected because `cliArgs.reporters` is _always_ not `nil` (it's an empty array instead). This change:

1. Accounted for that by checking `len` instead
2. Changed the behavior to be additive (merging the CLI reporters with the ones in the config file

File: cmd/captain/run.go (hunk 4 above)
Comment (kylekthompson) [reply cmd/captain/run.go:314]: See https://github.com/rwx-research/captain/pull/16#discussion_r1160715650 about the challenges of testing the reporter output 😐

Comment (kylekthompson) [reply cmd/captain/run.go:314]: In short, yes we should test this but exactly how I'm not quite sure yet since we want to run the old integration tests in a matrix and reporter output is subject to change (e.g. JSON/XML in non-breaking ways and the markdown if we want it to look different)

**Review by dan-manges — APPROVED** [review]
